code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,658 @@
+          description: Artifact version information
+        virtType:
+          type: string
+          enum:
+            - VM
+            - CONTAINER
+            - HELM 
+        fileName:
+          type: string
+          minLength:  8
+          maxLength: 32
+          description: Name of the file with the extension eg. zip, targz,etc.
+        repository:       
+          type: object
+          required:
+            - type
+            - url
+          properties:
+            type:                 
+              type: string
+              enum:
+                - PRIVATEREPO
+                - PUBLICREPO
+              description: Artifact repository location. PUBLICREPO is used of public urls like github, helm repo etc, 
+                                  PRIVATEREPO  is used for private repo managed by the application developer.
+            url:
+              $ref: '#/components/schemas/Uri'
+            userName:
+              type: string
+              pattern: ^[A-Za-z][A-Za-z0-9_]{7,63}$
+              description:  Username to acces the artifact repository
+            password:
+              type: string      
+              description:  Password to acces the artifact repository
+            token:
+              type: string          
+              description:  Authorization token acces the artifact repository   
+        componentSpec:
+          description: Details about compute, networking and storage requirements for each component of the application.  
+                              App provider should  define all information needed to instantiate the component. 
+                              If artifact is being defined at component level  this section should have information just about the component. 
+                              In case the artifact is being defined at application level  the section should provide details about all the components.
+          type: array
+          items:
+            type: object
+            required:
+              - componentName
+              - operatingSystem
+              - cpuArchitecture
+              - images
+              - networkInterfaces
+              - numOfInstances
+              - restartPolicy
+            properties:
+              componentName: 
+                type: string
+                pattern:  ^[A-Za-z0-9][A-Za-z0-9_]{6,62}[A-Za-z0-9]$
+                description: Must be a valid RFC 1035 label name.  Component name must be unique with an application
+              operatingSystem:
+                $ref: '#/components/schemas/OperatingSystem'
+              cpuArchitecture:
+                $ref: '#/components/schemas/CpuArchType'
+              images:
+                description: List of all images assosiated with the component. Images are uploaded or specified using UploadFile apis
+                type: array
+                items: 
+                  $ref: '#/components/schemas/FileId' 
+                minItems: 1
```

Comment:
There are no fileIds to associate with the component spec, as the file is actually part of the Artifact now. I suggest you remove the images field here, and change the component spec to a single ref instead of an array. That way it's clear there is a single component spec associated with the file that is specified in the Artifact.

Reply:
Agree, updated.